python/tank/log.py

```diff
@@ -557,9 +557,34 @@ def initialize_base_file_handler(self, log_name):
         :returns: The name of the previous log_name that is being switched away from,
                   None if no base logger was previously active.
         """
+        # avoid cyclic references
+        from .util import filesystem
+
+        return self.initialize_base_file_handler_from_path(
+            os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
+        )
+
+    def initialize_base_file_handler_from_path(self, log_file):
+        """
+        Create a file handler and attach it to the sgtk base logger.
+
+        This method is there for legacy Toolkit applications and shouldn't be used. Use
+        ``initialize_base_file_handler`` instead.
+
+        :param log_file: Path of the file to write the logs to.
+
+        :returns: The name of the previous log_name that is being switched away from,
```

Comment:
Hehe - fair enough! So to conclude - how does this sound?

- `initialize_base_file_handler(log_name)` - adjusted to return a full path to the previous log file that is being switched away from or None if no value was set.
- `initialize_base_file_handler_from_path(path)` also returns a full path in exactly the same manner as `initialize_base_file_handler()` above.
- `uninitialize_base_file_handler()` also adjusted to returns a full path like above
- we update the place in core where we make use of the `uninitialize_base_file_handler()` and `initialize_base_file_handler()` return values

Sounds right?